test/test_cuda.py

```diff
@@ -1289,11 +1289,21 @@ def test_grad_scaling_autocast_fused(self):
     # Compare non-fused optimizer vs fused one as the fused one unscales gradients
     # inside its cuda kernel unlike the other.
     def test_grad_scaling_autocast_fused_optimizers(self):
-        for optimizer_ctor, optimizer_kwargs, separate_unscale in product(
+        for optimizer_ctor, optimizer_kwargs, separate_unscale in list(product(
```

Comment:
Hmmm this makes me wonder...maybe one day these should live in test/test_optim.py as well...to deduplicate the configuration logic. (This is out of the scope of the PR)